src/bson.rs

```diff
@@ -434,10 +432,13 @@ impl Bson {
     #[doc(hidden)]
     pub fn to_extended_document(&self) -> Document {
         match *self {
-            Bson::RegExp(ref pat, ref opt) => {
+            Bson::RegExp(RegExp {
+                ref pattern,
+                ref options,
+            }) => {
                 doc! {
```

Comment:
When this implementation was originally written, legacy extended JSON was the only extended JSON :)

The BSON corpus tests have never been implemented for this library precisely because it's never been updated to the new extended JSON spec. If you're asking why we haven't implemented it since we took over stewardship of this library, then it's just due to having higher priority things to work on first.

Reply:
Seems like something we will want to fix for 1.0, but not in this PR. Filed [RUST-346](https://jira.mongodb.org/browse/RUST-346).

Reply:
Hmm, I thought we already had an auto-generated ticket for this, but I can't find it now.

Reply:
Yeah doesn't look like we had one. The only things coming up for "corpus" and "extended json" are the ticket I filed and decimal 128 support.